CONTRIBUTING.md

```diff
@@ -6,79 +6,177 @@ We are excited to see you want to be a part of this project by contributing. Her
+1. **Create a Branch**
+
+   ```bash
+   git checkout -b your-feature-branch  # Create a new branch for your feature
+   ```
 
-A special note. If there will be changes to the [app config](https://github.com/janus-idp/backstage-showcase/blob/main/app-config.yaml), we ask that [documentation](https://github.com/janus-idp/backstage-showcase/blob/main/showcase-docs/getting-started.md) be updated if it will be a requirement for running the app. We also ask to ensure that the app will still work in the case of dummy information being supplied to the app config. While it is not a hard requirement, it does help others with quickly being able to get up and running with the showcase app.
+2. **Implement Your Changes**
 
-### Adding statically linked plugins for frontend and backend
+   Make your code changes, ensuring that you follow the project's coding standards and best practices.
+
+3. **Testing**
+
+   - **Run Tests:** Ensure all tests pass before committing.
+
+     ```bash
+     yarn test         # Run unit tests
+     cd e2e-tests
+     yarn showcase     # Run e2e tests
+     ```
+
+     _For more on the e2e, check the [e2e contributing guidelines](./docs/e2e-tests/CONTRIBUTING.MD)_
+
+   - **Note on Environment Dependencies:**
+
+     - If your new or edited code can't be validated locally due to environment dependencies, you can open a **draft Pull Request (PR)**. This allows the tests to run on the test environment as described in our CI documentation.
+
+4. **Linting and Formatting**
+
+   Ensure your code passes linting and formatting checks.
+
+   ```bash
+   yarn lint:check     # Checks for linting errors
+   yarn lint:fix       # Fixes linting errors automatically
+   yarn prettier:check # Checks for formatting issues
+   yarn prettier:fix   # Fixes formatting issues automatically
+   ```
 
-When contributing a new @internal plugin into this repo, you must remember to add the plugin to the Dockerfiles under the section titled `Stage 2 - Install dependencies`:
+5. **Ensure CI Passes**
+
+   Your contributions will need to pass the Continuous Integration (CI) tests for pull requests.
+
+6. **Commit Changes**
+
+   Use meaningful commit messages following the [Conventional Commits](https://www.conventionalcommits.org/) specification.
+
+   ```bash
+   git commit -m "feat: add new feature"  # Commit your changes with a meaningful message
+   ```
+
+7. **Update Documentation**
+
+   If there will be changes to the [app config](https://github.com/janus-idp/backstage-showcase/blob/main/app-config.yaml), we ask that [documentation](https://github.com/janus-idp/backstage-showcase/blob/main/showcase-docs/getting-started.md) be updated if it will be a requirement for running the app. We also ask to ensure that the app will still work in the case of dummy information being supplied to the app config. While it is not a hard requirement, it does help others with quickly being able to get up and running with the showcase app.
```

Comment:
Should these links to files in the repo be relative instead?  Looks like it's been this way for several of these but I wonder if it'd be better for tooling if these are referring to the files the URLs are pointing to.  I guess if this is rendered in techdocs these would bring you out of there and onto github for example.

Reply:
totally. Also, just find out that `getting-started.md` does not exist anymore 🙁